pkg/R/sparkR.R

```diff
@@ -149,11 +149,15 @@ sparkR.init <- function(
     }
     f <- file(path, open='rb')
     sparkRBackendPort <- readInt(f)
+    monitorPort <- readInt(f)
     close(f)
     file.remove(path)
-    if (length(sparkRBackendPort) == 0) {
+    if (length(sparkRBackendPort) == 0 || sparkRBackendPort == 0 ||
+     length(monitorPort) == 0 || monitorPort == 0) {
       stop("JVM failed to launch")
     }
+    # never close this socket, JVM is waiting for it
+    .sparkREnv$monitorF <- socketConnection(port = monitorPort)
```

Comment:
So the design we are using in SparkR is that the JVM backend is tied one-to-one with the SparkContext. So the expected behavior at `sparkR.stop` is to stop the JVM backend (This could be necessary for any reason: testing or even just reducing memory used in the system etc.)

In unit tests we never actually call `sparkR.stop` -- we just call `sparkR.init` and that reuses the SparkContext across tests. BTW I'll continue this comment in the main PR thread as it'll be easier to follow there.